From 8f833f673bc36de18835fe6cd6ff84ab0e2fe7a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Erdmann?= <dywi@mailerd.de>
Date: Sat, 26 Jan 2013 17:22:48 +0100
Subject: [PATCH] tlp-functions: fix get_wifi_ifaces()

This commit fixes a case where get_wifi_ifaces() returns '*',
which results in any file/dir in $PWD being treated as wifi interface.
Only occurs if the system has no wireless devices.
---
 tlp-functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tlp-functions b/tlp-functions
index 9966c19..76de39c 100755
--- a/tlp-functions
+++ b/tlp-functions
@@ -815,7 +815,7 @@ get_wifi_ifaces () { # get all wifi devices -- retval: $wifaces
     wifaces=""
     
     for wi in /sys/class/net/*/wireless; do
-        if [ -n "$wi" ]; then
+        if [ -e "$wi" ]; then
             wi=${wi%/wireless}; wi=${wi##*/}
             
             [ -n "$wifaces" ] && wifaces="$wifaces "
-- 
1.8.1.1

